### DISCUSSION TITLE
Install from original media

### DISCUSSION THREAD

**Discussion by farlies** [DescentDevelopers/Descent3#273] [Q&A] [answered]
I have a pair of original CD-ROMs purchased and enjoyed, uh, quite a long time ago.  Is it possible to install game files from these disks onto Ubuntu (mantic/numbat)?  Any tips?  Or is this a waste of time?  Already have Steam/Wine on this system.

**Comment by JeodC** [comment]
Are they windows CDs or the Loki linux CDs?

**Reply by farlies** [reply]
Windows CDs (from when the game was fairly new.)

**Reply by JeodC** [reply]
There may be a tool to extract files from the .cabs, I can look into it later.

**Reply by JeodC** [reply]
Use the cabextract tool.

**Reply by Jayman2000** [reply]
> Use the cabextract tool.

I don’t think that cabextractg is the right tool for the job. [Its Web site says:](https://www.cabextract.org.uk/#unshield)

<blockquote>
<b>How can I extract this <samp>DATA1.CAB</samp> file?</b>
 <p>There are two different "cabinet" file formats in popular use. Some are Microsoft cabinets, which can be unpacked with cabextract. Others are InstallShield cabinets, which come as files <samp>data1.hdr</samp>, <samp>data1.cab</samp>... Unpacking an InstallShield cabinet with cabextract gives the message <samp>WARNING; found InstallShield header. This is probably an InstallShield file</samp>. InstallShield cabinet files can be unpacked with <a href="https://github.com/twogood/unshield"><b>unshield</b></p>
</blockquote>

When I try to run `cabextract` on Descent 3 CD 1’s `data1.cab` file, it fails with the same error that the Web site mentions. When I try to run `unshield` on the same file, it works.

**Comment by Jayman2000** [comment, marked as answer]
I recently submitted new test results for [running the CD version of Descent 3 in Wine](https://appdb.winehq.org/objectManager.php?sClass=version&iId=884). In those test results, I included an in-depth installation guide. Unfortunately, my test results need to reviewed before they appear on winehq.org, so I’ll post the installation guide that I wrote here:

<h2>Step by step installation procedure for the Dual-Jewel version of Descent 3</h2>
<ol><li><em>(Recommended)</em> Run <code>winecfg</code> and make sure that “Emulate a virtual desktop” is enabled.</li><li><em>(Optional)</em> Determine if you’re going to be affected by a bug with Descent 3’s installer, and potentially apply a workaround:<ol type="a"><li>Download&nbsp;<a href="https://codeberg.org/JasonYundt/environment-size-checker" target="_blank">Environment Size Checker</a>.</li><li>Run <code>wine environment-size-checker.exe</code></li><li>If that program tells you that your environment is more than 32,724 bytes large, then you’ll need to unset or shorten environment variables before running Descent 3’s installer. If you don’t, then the installer will page fault.</li></ol></li><li>Install Descent 3:<ol type="a"><li>Insert disc 1.</li><li>Make sure that disc 1 is mounted.</li><li>Determine which drive letter Wine is using for your CD drive. (Hint: try running <code>wine explorer</code>).</li><li>Run <code>wine '&lt;drive-letter&gt;:\Setup.exe'</code></li><li>Follow the installation wizard’s instructions until it asks you to insert disc 2.</li><li>Switch to disc 2:<ol type="I"><li>Run <code>wine eject &lt;drive-letter&gt;:</code></li><li>Make sure that the disc was unmounted and ejected.</li><li>Insert disc 2.</li><li>Mount disc 2.</li></ol></li><li>Follow the installation wizard’s instructions until it asks you to insert disc 1 again.</li><li>Switch back to disc 1. Follow a similar procedure to the one that you used to switch to disc 2.</li><li>Finish the installation wizard.</li><li>When the installation wizard finishes, it will open an explorer window. Close out of that window.</li><li>Unmount the disc.</li><li>Eject the disc.</li></ol></li><li>Install Descent 3: Mercenary:<ol type="a"><li>Insert disc 3.</li><li>Make sure that disc 3 is mounted.</li><li>Run <code>wine start /d &lt;drive-letter&gt;: setup.exe -autorun</code>. WARNING: If you forget to use the <code>-autorun</code> option, then you’ll have to manually create registry entries later.</li><li>Follow the instructions in the installation wizard.</li></ol></li><li><p>Install&nbsp;<a href="http://www.descent3.com/downloads.php" target="_blank">the v1.4 patch</a>.</p><p><b>Note:</b> if the v1.4 patch isn’t working for you, then you may need to use the Dual-Jewel fix. For more information, look for  <a href="http://descent3.com/faq.php">this question in descent3.com’s FAQ: “Q: I have the Dual-Jewel version of Descent 3, and when I try to install any of the patches, the installation terminates with an error message stating: 'Old File Not Found'. What do I do?” </a>.</p></li><li><em>(Optional)</em> Install&nbsp;<a href="http://www.descent3.com/downloads.php">the NOCD patch</a>.</li><li>If you didn’t to use the <code>-autorun</code> option while installing Descent 3: Mercenary, then you’ll have to create a registry entry manually. See <a href="https://www.descentbb.net/viewtopic.php?p=391548#p391548" target="_blank">this forum post</a> for details.</li></ol>

**Reply by farlies** [reply]
Worked like a charm. Instructions could hardly be clearer.  Thanks to the whole crew for your noble work.
